src/zkevm_specs/evm/execution/block_coinbase.py

```diff
@@ -0,0 +1,19 @@
+from ..instruction import Instruction, Transition
+from ..table import BlockContextFieldTag
+from ..opcode import Opcode
+
+
+def coinbase(instruction: Instruction):
+    opcode = instruction.opcode_lookup(True)
+    instruction.constrain_equal(opcode, Opcode.COINBASE)
+    address = instruction.stack_push()
+    # in real circuit also check address raw data is 160 bit length (20 bytes)
+    # check block table for coinbase address
+    instruction.constrain_equal(address, instruction.block_lookup(BlockContextFieldTag.Coinbase))
```

Comment:
If we want to keep field `coinbase` in table as actual value instead of random linear combination, we need to do a base conversion here (from `256` to `r`).

```suggestion
    instruction.constrain_equal(
        address,
        instruction.bytes_to_rlc(instruction.int_to_bytes(instruction.block_lookup(BlockContextFieldTag.Coinbase), 20)),
    )
```

Reply:
fixed in 164f7de. thanks